dtls.c

```diff
@@ -4122,7 +4122,17 @@ dtls_handle_message(dtls_context_t *ctx,
       if (err < 0) {
 	dtls_warn("error while handling handshake packet\n");
 	dtls_alert_send_from_err(ctx, peer, session, err);
-	return err;
+
+        if (peer && err < -(1 << 8) && err > -(3 << 8)) {
+          if (DTLS_ALERT_LEVEL_FATAL == ((-err) & 0xff00) >> 8) {
```

Comment:
It seems to me that this check is essentially the same as the check for `err` in the previous line. As this one is more explicit in what is really being checked, I suggest keeping this one and get rid of the other.

Reply:
Updated and code pushed.